### DISCUSSION TITLE
Whats the rule for creating the documentation URL of a check?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [checkstyle/checkstyle#13751] [General]
I'm currently working on providing online help for violations in EclipseCS. For that, I'm calculating the URL dynamically from the checkstyle metadata. E.g. in the sources all paths like checks/javadoc/JavadocMethodCheck.xml can easily be converted to the matching URL https://checkstyle.org/checks/javadoc/javadocmethod.html#JavadocMethod etc., but for the "misc" category this isn't true, since there is no misc in the sources.
Anyone who knows the scheme for mapping checks to URLs?

**Comment by rnveach** [comment]
Just to warn, the URL pathing you shown above is something that was just recently introduced and is still pretty new. I don't think it plans to change soon, but this makes more sense to me to request this as an additional parameter in the metadata that we provide a link to our own documentation. Our meta doesn't include our examples, so this makes a link to documentation more useful.

To answer your question as best I can, the URL to the misc section is https://checkstyle.org/checks/misc/index.html . If your trying to identify which checks are misc and which aren't, there isn't anything concrete. For example, it looks like majority of misc checks are located in https://github.com/checkstyle/checkstyle/tree/master/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks , but indentation isn't stored in that area which are also misc. They are located in https://github.com/checkstyle/checkstyle/tree/master/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/indentation . I am not seeing anything in the XML file themselves that clearly says "This is misc and not its own category".

You could get away with using the path in `fully-qualified-name`, but like I said, it isn't really a guarantee. Hence why I think requesting the URL (and maybe the category to be the better option).

**Reply by romani** [reply]
We probably can move indentation under misc and it will predictable, right?

I think we can do this easily. @rnveach , do you agree to move it ?

**Reply by rnveach** [reply]
I don't personally like the non-folder for misc. Again, I think it is better we add more information to the meta information.

**Reply by romani** [reply]
I am confused, 
https://checkstyle.org/checks/misc/indentation.html is under misc.

But conceptually should be in its own group indentation
https://github.com/checkstyle/checkstyle/blob/d94f78fffde2b3180b911c6d8f8c79250803cb9d/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/indentation/IndentationCheck.xml#L4

**Reply by romani** [reply]
During a long time nobody joined this package.
We can keep it as exception in plugin for proper link generation

**Reply by Bananeweizen** [reply]
I'd also prefer having this in the metadata itself, because that way all tools could eventually rely on that information. The only backside would be to defer the feature in Eclipse until we are on a recent version containing that feature (while my implementation is basically ready now). Until then I could go with a rule of thumb implementation only, that might fail in some cases.

**Reply by romani** [reply]
Please Do not wait , do special handling for indentation and move forward.
